cirq/study/resolver_test.py

```diff
@@ -13,33 +13,102 @@
 # limitations under the License.
 
 """Tests for parameter resolvers."""
+import fractions
 
 import numpy as np
+import pytest
 import sympy
 
 import cirq
 
 
-def test_value_of():
+@pytest.mark.parametrize('val', [
+    3.2,
+    np.float32(3.2),
+    int(1),
+    np.int(3),
+    np.int32(45),
+    np.float64(6.3),
+    np.int32(2),
+    np.complex64(1j),
+    np.complex128(2j),
+    np.complex(1j),
+    fractions.Fraction(3, 2),
+])
+def test_value_of_pass_through_types(val):
+    _assert_consistent_resolution(val, val)
+
+
+@pytest.mark.parametrize('val,resolved', [(sympy.pi, np.pi),
+                                          (sympy.S.NegativeOne, -1),
+                                          (sympy.S.Half, 0.5),
+                                          (sympy.S.One, 1)])
+def test_value_of_transformed_types(val, resolved):
+    _assert_consistent_resolution(val, resolved)
+
+
+@pytest.mark.parametrize('val,resolved', [(sympy.I, 1j)])
+def test_value_of_substituted_types(val, resolved):
+    _assert_consistent_resolution(val, resolved, True)
+
+
+def _assert_consistent_resolution(v, resolved, subs_called=False):
```

Comment:
Would it be worth having a short docstring here or in SubsAwareSymbol explaining why we have "subs_called" and why we are trying to avoid it?

Reply:
I'll add some explanation both here and to the helper class.